File: 1445-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.py
class Solution(object):
    def numOfSubarrays(self, arr, k, threshold):
        
        window = 0
        for i in range(k):
            window += arr[i]
        
        count = 0
        if window/k >= threshold:
            count += 1
        
        start = 0
        end = k
        while end < len(arr):

            window -= arr[start]
            start += 1

            window += arr[end]
            end += 1

            if window/k >= threshold:
                count += 1

        return count        

File: 1445-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.java
class Solution {
    public int numOfSubarrays(int[] arr, int k, int threshold) {
        int window = 0;
        for(int i=0; i<k; i++){
            window += arr[i];
        }

        int count = 0;
        int avg = window/k;

        if(avg >= threshold)count++;

        for(int i=k; i<arr.length; i++){
            window -= arr[i-k];
            window += arr[i];
            if(window/k >= threshold)count++;
        }
        return count;
    }
}